### ISSUE TITLE
Allow pre-releases to be created without modifying the base version

### ISSUE THREAD

**Issue by StephanBijzitter** [webpro/release-it#56]
When I push a project with version `1.2.3`, I have configured my CI to automatically pre-release it as `1.2.4-commithash.0`, this is because I use `release-it prepatch`. I would prefer to create the release, without modifying the patch version, resulting in `1.2.3-commithash.0`.

**Comment by webpro** [comment]
Maybe you can create a small script that resolves the desired version, and then run release-it with that.

**Comment by StephanBijzitter** [comment]
True, but it'd still be nice to have it included in this library :-)

**Comment by msegado** [comment]
The core issue here is that node-semver (and by extension, release-it) only provides an API to _increment_ the version, but `1.2.3-commithash.0` is a version _decrement_ compared to `1.2.3`. There's an undocumented `pre` command in node-semver that does exactly what you want, but per my comment in #59 its probably not safe to expose or document it in release-it (at least not without pinning the exact patch version of node-semver in `package.json`) since it could hypothetically vanish without notice...

**Comment by StephanBijzitter** [comment]
Well, although you are right that it in fact decrements the version, this is only the case when the `1.2.3` version already exists. In our project, this is how we work:

`master` is stable, and as such any changes to any module or additions of new modules, must be tested before merging to master. To do this, we work from branches where every commit is released using release-it under a beta flag. What happens now is the following, assuming an example for a new module that is yet to be released: `flying-dutchman`.

* Developers checkouts out `master`
* Developers creates new branch `module-flying-dutchman`
* Developers creates the module and gives it version `1.0.0`
* Developer pushes branch to origin
* CI releases a beta of this module as `1.0.1-commithash.0`
* Tester tests the beta release in a testing environment
* Developer 2 merges pull request into master
* CI detects change in master and releases this module as `1.0.0`

In other words, the beta release of 1.0.1 was released before the actual release of 1.0.0, which is a bit odd.
